src/powerbox/tools.py

```diff
@@ -74,6 +83,26 @@ def angular_average(
     log_bins : bool, optional
         Whether to create bins in log-space.
 
+    interpolation_method : str, optional
+        If None, does not interpolate. Currently only 'linear' is supported.
+
+    interp_points_generator : callable, optional
```

Comment:
Here you need to define what arguments the `inter_points_generator` should take, and what it needs to return. Users need to know how to write such a function.